Sources/SourceControl/GitRepository.swift

```diff
@@ -245,6 +258,25 @@ public class GitRepository: Repository, WorkingCheckout {
         }
     }
 
+    public func remove() throws {
```

Comment:
I think it makes more sense for the Repository to expose an API which just says if the repository has edits, and for the higher level client to have the actual `remove` API. I think of `Repository` as more or less exposing the same things you do with `git`, so `remove` here should mean removal of a file in the repository.